File: Python/125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l = 0
        r = len(s) - 1
        while l < r:
            while not ('0' <= s[l] <= '9' or 'a' <= s[l] <= 'z' or 'A' <= s[l] <= 'Z') and l < r:
                l += 1
            while not ('0' <= s[r] <= '9' or 'a' <= s[r] <= 'z' or 'A' <= s[r] <= 'Z') and l < r:
                r -= 1            
            if s[l].lower() != s[r].lower():
                return False
            l += 1
            r -= 1
        return True

File: 2024/125. Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i = 0, j = s.length() - 1;
        while (i < j) {
            while (i < j && !((s[i] >= 'A' && s[i] <= 'Z') || (s[i] >= 'a' && s[i] <= 'z') || (s[i] >= '0' && s[i] <= '9'))) {
                i++;
            }
            if (i < j && s[i] >= 'A' && s[i] <= 'Z') {
                s[i] = s[i] - 'A' + 'a';
            }
            while (i < j && !((s[j] >= 'A' && s[j] <= 'Z') || (s[j] >= 'a' && s[j] <= 'z') || (s[j] >= '0' && s[j] <= '9'))) {
                j--;
            }
            if (i < j && s[j] >= 'A' && s[j] <= 'Z') {
                s[j] = s[j] - 'A' + 'a';
            }
            if (i < j && s[i] != s[j]) {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};